CRM/Utils/System/Drupal.php

```diff
@@ -163,7 +163,10 @@ public static function checkUserNameEmailExists(&$params, &$errors, $emailName =
    */
   public function getLoginURL($destination = '') {
     $query = $destination ? array('destination' => $destination) : array();
-    return url('user', array('query' => $query), TRUE);
+   	$config = CRM_Core_Config::singleton();
```

Comment:
I think it could be improved with $baseUrl = CRM_Core_Config::singleton()->userFrameworkBaseURL;